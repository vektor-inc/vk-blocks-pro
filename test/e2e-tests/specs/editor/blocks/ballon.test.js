/**
 * WordPress dependencies
 */
 import {
	clickBlockAppender,
	getEditedPostContent,
	createNewPost,
} from '@wordpress/e2e-test-utils';

describe( 'Ballon', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'can be created by typing "/html"', async () => {
		// Create a Custom HTML block with the slash shortcut.
		await clickBlockAppender();
		await page.keyboard.type( '/html' );
		await page.waitForXPath(
			`//*[contains(@class, "components-autocomplete__result") and contains(@class, "is-selected") and contains(text(), 'Custom HTML')]`
		);
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( '<p>Pythagorean theorem: ' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type(
			'<var>a</var><sup>2</sup> + <var>b</var><sup>2</sup> = <var>c</var><sup>2</sup> </p>'
		);

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );