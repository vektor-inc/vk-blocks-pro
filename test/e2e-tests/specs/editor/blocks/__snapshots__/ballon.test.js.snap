// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ballon can be created by typing "/html" 1`] = `
"<!-- wp:html -->
<p>Pythagorean theorem: 
<var>a</var><sup>2</sup> + <var>b</var><sup>2</sup> = <var>c</var><sup>2</sup> </p>
<!-- /wp:html -->"
`;
